File: app/Models/Driver.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Laravel\Sanctum\HasApiTokens;

class Driver extends Model
{
    use HasApiTokens, HasFactory;

    protected $fillable = [
        'name', 'phone', 'email', 'google_id', 'fcm_token', 'photo_url', 'rating', 'total_trips', 'completion_rate',
        'years_experience', 'identity_verified', 'licence_verified', 'background_checked',
        'languages', 'operating_area', 'member_since', 'verification_status', 'preferred_locale',
    ];

    protected function casts(): array
    {
        return [
            'languages' => 'array',
            'member_since' => 'date',
            'identity_verified' => 'boolean',
            'licence_verified' => 'boolean',
            'background_checked' => 'boolean',
            'rating' => 'decimal:2',
            'completion_rate' => 'decimal:2',
        ];
    }

    public function isFullyVerified(): bool
    {
        return $this->identity_verified && $this->licence_verified && $this->background_checked;
    }

    public function vehicles(): HasMany
    {
        return $this->hasMany(Vehicle::class);
    }

    public function documents(): HasMany
    {
        return $this->hasMany(DriverDocument::class);
    }

    public function bookings(): HasMany
    {
        return $this->hasMany(Booking::class);
    }

    public function subscription(): HasOne
    {
        return $this->hasOne(DriverSubscription::class)->latestOfMany();
    }

    public function reviews(): HasMany
    {
        return $this->hasMany(Review::class);
    }
}

File: taxiway/lib/core/models/driver.dart
import 'review.dart';

class Driver {
  final String id;
  final String name;
  final String photoUrl;
  final double rating;
  final int totalTrips;
  final int completionRate;
  final double yearsExperience;
  final bool identityVerified;
  final bool licenceVerified;
  final bool backgroundChecked;
  final String phone;
  final List<String> languages;
  final String operatingArea;
  final String memberSince;

  /// Populated once, when this driver was fetched for a specific booking
  /// (see Api\Customer\DriverController@show) — empty otherwise.
  final List<Review> reviews;

  const Driver({
    required this.id,
    required this.name,
    required this.photoUrl,
    required this.rating,
    required this.totalTrips,
    required this.completionRate,
    required this.yearsExperience,
    required this.identityVerified,
    required this.licenceVerified,
    required this.backgroundChecked,
    required this.phone,
    required this.languages,
    required this.operatingArea,
    required this.memberSince,
    this.reviews = const [],
  });

  bool get isFullyVerified => identityVerified && licenceVerified && backgroundChecked;

  factory Driver.fromJson(Map<String, dynamic> json) {
    final phone = json['phone'] as String;
    final memberSinceRaw = json['member_since'] as String?;

    return Driver(
      id: json['id'].toString(),
      name: json['name'] as String,
      photoUrl: json['photo_url'] as String? ?? '',
      rating: double.parse(json['rating'].toString()),
      totalTrips: json['total_trips'] as int,
      completionRate: double.parse(json['completion_rate'].toString()).round(),
      yearsExperience: double.parse(json['years_experience'].toString()),
      identityVerified: json['identity_verified'] as bool,
      licenceVerified: json['licence_verified'] as bool,
      backgroundChecked: json['background_checked'] as bool,
      phone: phone.length == 10 ? '+91 ${phone.substring(0, 5)} ${phone.substring(5)}' : phone,
      languages: (json['languages'] as List?)?.map((e) => e.toString()).toList() ?? const [],
      operatingArea: json['operating_area'] as String? ?? '',
      memberSince: memberSinceRaw != null ? DateTime.parse(memberSinceRaw).year.toString() : '',
      reviews: (json['reviews'] as List?)?.map((e) => Review.fromJson(e as Map<String, dynamic>)).toList() ?? const [],
    );
  }
}
